### DISCUSSION TITLE
Update a widget value from another script?

### DISCUSSION THREAD

**Discussion by svandragt** [beeware/toga#1325] [Q&A] [answered]
Hello, say I'm building an interface with a readonly MultilineTextInput. Now I want to add lines to the widget from another part of the application. Currently I do this:

app.py:
```
"""
test
"""
import toga

me = None


class MyApp(toga.App):
    def add_log(self, text):
        self.logbox.value += text
        self.logbox.refresh()

    def startup(self):
        box = toga.Box(style=Pack(direction=COLUMN))
        self.logbox = toga.MultilineTextInput(id='logbox', readonly=True)
        self.logbox.style.flex = 1
        box.add(self.logbox)

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = box
        self.main_window.show()


def main():
    global me
    me = MyApp()
    return me
```

and in another part of the app:
```
from . import app

def example():
  app.me.add_log('Hello World')
```

Is that the best practice way? I was unable to find any example in the docs that updated a widget from outside the app class. As the widgets can all have an `id` should I be able to call a widget to update by their `id` for example? Thanks for your time.

**Comment by freakboy3742** [comment, marked as answer]
There will only ever be a single App instance in a running executable, and the app tracks that instance as `App.app`; so the following will do what you need:
```
import toga

def example():
    toga.App.app.add_log('Hello World')
```

However, generally speaking, I'd advise against relying on `App.app`. It works (and needs to exist for a variety of reason); but as a general matter of code style, if there's "another part of your app" that needs to know about the app, then it should be constructed and configured with a reference *to* the app (or, preferably, with a reference to a common data source that, when updated, is reflected in the app's GUI).
